packages/contracts-bedrock/test/setup/Setup.sol

```diff
@@ -92,7 +92,13 @@ contract Setup {
             hex"7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe03601600081602082378035828234f58015156039578182fd5b8082525050506014600cf3"
         );
 
-        deploy.run();
+        string memory allocsPath = string.concat(vm.projectRoot(), "/.testdata/genesis-l1.json");
+        if (vm.isFile(allocsPath)) {
```

Comment:
since genesis-l1.json is now a requirement for tests, shouldn't we bail early here if it's missing?

Reply:
You are right that this is weird and gives a footgun. I didn't want to force devs to run the out of process build script on every update to the smart contracts, with this you can do `forge test ...` when the artifacts are not present and have the up to date solidity code in the tests.

This is a temp fix, happy to change it if you think that its too big of a footgun, its a tradeoff in devex. This whole thing is going to be replaced soon with L2 genesis being built in solidity.

Reply:
gotcha. fwiw, I do run `forge test` rather than `pnpm` for test filtering. But the rationale makes sense as a temporary tradeoff.